### PR TITLE
Made-JS Feature/SOF-5098

### PR DESCRIPTION
**Pull request by adewyer**

### CODE REVIEW

File: src/material.js [hunk 1]
```diff
@@ -99,6 +99,12 @@ export class Material {
         return this.prop('isNonPeriodic', false, true);
     }
 
+    /**
+     * @summary Sets the value of isNonPeriodic based on Boolean value passed as an argument.
+     * @param bool
+     */
+    updateIsNonPeriodic(bool) { this.prop('isNonPeriodic', bool);}
```
Comment (timurbazhirov) [inline src/material.js:106]: This doesn't update anything actually.

See like 121 of this file for how the update should be done

Comment (adewyer) [reply src/material.js:106]: Yes, sorry this got updated in a later PR. I will update here as well.

File: src/material.js [hunk 2]
```diff
@@ -99,6 +99,12 @@ export class Material {
         return this.prop('isNonPeriodic', false, true);
     }
 
+    /**
+     * @summary Sets the value of isNonPeriodic based on Boolean value passed as an argument.
+     * @param bool
```
Comment (timurbazhirov) [inline src/material.js:104]: `{Boolean} bool`

**Review by timurbazhirov — APPROVED** [review]

File: src/material.js [hunk 3]
```diff
@@ -80,7 +80,8 @@ export class Material {
         return {
             lattice: this.Lattice.toJSON(),
             basis: this.Basis.toJSON(),
-            name: this.name || this.formula
+            name: this.name || this.formula,
+            isNonPeriodic: this.isNonPeriodic || false
```
Comment (adewyer) [inline src/material.js:84]: This function fixes the cloning issue inside of https://github.com/Exabyte-io/materials-designer/pull/29/files#diff-07117e6feb13357543259d645dcae43645f34789d559cfebdcd62a99e17c3604

**Pull request merged**